cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala

```diff
@@ -36,22 +38,25 @@ object Formatter {
 
   val allFormatters: Set[FormatterCompanion] = Set(defaultFormatter, Json, Sarif)
 
-  def apply(formatterName: String,
+  def apply(outputConfiguration: CLIConfiguration.Output,
             executionDirectory: File,
-            outputFile: Option[File] = Option.empty,
             printStream: Option[PrintStream] = Option.empty): Formatter = {
 
-    val builder = allFormatters.find(_.name.equalsIgnoreCase(formatterName)).getOrElse {
-      logger.warn(s"Could not find formatter for name $formatterName. Using ${defaultFormatter.name} as fallback.")
+    val stream = outputConfiguration.file.map(asPrintStream).orElse(printStream).getOrElse(defaultPrintStream)
+    val formatterConfiguration = Configuration(stream, executionDirectory, outputConfiguration.ghCodeScanningCompat)
+
+    val formatterBuilder = allFormatters.find(_.name == outputConfiguration.format).getOrElse {
+      logger.warn(
+        s"Could not find formatter for name ${outputConfiguration.format}. Using ${defaultFormatter.name} as fallback.")
       defaultFormatter
     }
 
-    val stream = outputFile.map(asPrintStream).orElse(printStream).getOrElse(defaultPrintStream)
-
-    builder(stream, executionDirectory)
+    formatterBuilder(formatterConfiguration)
   }
 
   private def asPrintStream(file: File) = {
     new PrintStream(new FileOutputStream(file.toJava, false))
   }
+
+  case class Configuration(printStream: PrintStream, executionDirectory: File, ghCodeScanningCompat: Boolean)
```

Comment:
Now that I'm looking at the diff I'm not sure if there is any advantage to this wrapping.